lib/twilio/audioplayer/deferred.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @packageDocumentation
+ * @module Voice
+ * @internalapi
+ */
+// @ts-nocheck
+export default class Deferred<T> {
+  readonly promise: Promise<T>;
+
+  private _reject: (e?: any) => void;
+  get reject() { return this._reject; }
+
+  private _resolve: (t?: T) => void;
+  get resolve() { return this._resolve; }
+
+  constructor() {
+    this.promise = new Promise((resolve, reject) => {
```

Comment:
Can you pass the type parameter to the Promise here so we don't have to typecast the `Deferred` objects in the main class?